wdl-lint/src/rules/malformed_lint_directive.rs

```diff
@@ -0,0 +1,103 @@
+//! A lint rule for flagging malformed lint directives.
+use wdl_ast::AstToken;
+use wdl_ast::Comment;
+use wdl_ast::Diagnostic;
+use wdl_ast::Diagnostics;
+use wdl_ast::Document;
+use wdl_ast::Span;
+use wdl_ast::SupportedVersion;
+use wdl_ast::SyntaxElement;
+use wdl_ast::SyntaxKind;
+use wdl_ast::ToSpan;
+use wdl_ast::VisitReason;
+use wdl_ast::Visitor;
+use crate::Rule;
+use crate::Tag;
+use crate::TagSet;
+use crate::util::is_inline_comment;
+
+/// The identifier for the Malformed Lint Directive rule.
+const ID: &str = "MalformedLintDirective";
+/// The accepted lint directives.
+const ACCEPTED_LINT_DIRECTIVES: [&str; 1] = ["except"];
+
+/// Creates an "Inline Lint Directive" diagnostic.
+fn inline_lint_directive(
+    span: Span,
+) -> Diagnostic {
+    Diagnostic::warning(
+        "lint directive must be on its own line")
+        .with_rule(ID)
+        .with_label("malformed lint directive", span,
+        )
+        .with_fix("place lint directive on new line")
+}
+
+/// Creates an "Invalid Lint Directive" diagnostic.
+fn invalid_lint_directive(
+    span: Span,
+) -> Diagnostic {
+    let accepted_directives = ACCEPTED_LINT_DIRECTIVES.join(", ");
+    Diagnostic::warning(
+        "lint directive not recognized")
+        .with_rule(ID)
+        .with_label("lint directive not recognized", span)
+        .with_fix(format!("remove unrecognized lint directives, use any of the directives included: [{:#?}]", accepted_directives))
```

Comment:
Can you respond with the exact error you're seeing? This looks wrong to my eye, but can't put my finger on what the issue is without the error messaging.

Reply:
The error message I get it: 
```
---- visitor::test::it_supports_reuse stdout ----
thread 'visitor::test::it_supports_reuse' panicked at wdl-lint/src/visitor.rs:411:14:
should not have any diagnostics: [Diagnostic { rule: Some("MalformedLintDirective"), severity: Warning, message: "lint directive not recognized", fix: Some("remove unrecognized lint directives, use any of the directives included: [\"except\"]"), labels: [Label { message: "lint directive not recognized", span: Span { start: 15, end: 53 } }] }]
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace


failures:
    visitor::test::it_supports_reuse
```
Previously, I have encountered this error when I wasn't safely handling an Option, but now I'm a bit turned around on what's causing it. My hunch is that the `invalid_lint_directive()` is causing the error, because commenting out code and re-testing, the problem appears with this code active.

Reply:
It's not very obvious what's happening here, but the problem is that you're new rule is adding a diagnostic on the [WDL document created by `it_supports_reuse`](https://github.com/stjude-rust-labs/wdl/blob/main/wdl-lint/src/visitor.rs#L392-L399). I'm not sure why that is, probably some buggy logic in your detection of an invalid lint directive (because the lint directive in that document looks fine).

The `it_supports_resuse` test assumes that the WDL snippet I linked is lint-diagnostic-free, and your new rule is inserting one.

Reply:
(That is a kind of an annoying assumption for the test to make and we should consider finding a different way to write this test)

Reply:
@a-frantz I hope I'm not misinterpreting the presence of these errors, but where should I go from here?

Reply:
Lines 90 and 91 aren't doing what you expect them to, so the following `add()` is inserting a diagnostic in `it_supports_reuse` when one isn't expected. Rework those two lines until this test stops failing.